Enable prescribing 3d fields from files

LenkaNovak opened the pull request:
Closes https://github.com/CliMA/ClimaCoupler.jl/issues/507

<!---
Review checklist

I have:
- followed the codebase contribution guide: https://clima.github.io/ClimateMachine.jl/latest/Contributing/
- followed the style guide: https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/
- followed the documentation policy: https://github.com/CliMA/policies/wiki/Documentation-Policy
- checked that this PR does not duplicate an open PR.

In the Content, I have included 
- relevant unit tests, and integration tests, 
- appropriate docstrings on all functions, structs, and modules, and included relevant documentation.

-->

----
- [x] I have read and checked the items on the review checklist.

juliasloan25 commented on src/Regridder.jl:
```diff
@@ -64,6 +65,33 @@ function reshape_cgll_sparse_to_field!(field::Fields.Field, in_array::Array, R)
     Spaces.dss!(Fields.field_values(field), topology, hspace.quadrature_style)
 end
 
+function reshape_cgll_sparse_to_field!(
```
could you add comments explaining that this method is for 3d spaces and the one above is for 2d?

juliasloan25 commented on src/Regridder.jl:
```diff
@@ -64,6 +65,33 @@ function reshape_cgll_sparse_to_field!(field::Fields.Field, in_array::Array, R)
+    Nf = size(field_array, 4)
+    Nz = size(field_array, 1)
+
+    for z in 1:Nz
```
and also maybe a comment here saying that this is the main difference between the 2d and 3d cases

juliasloan25 commented on test/Project.toml:
```diff
@@ -36,6 +36,7 @@ Unitful = "1986cc42-f94f-5a68-af5c-568840ba703d"
 ArtifactWrappers = "0.2"
 Aqua = "0.8"
 ClimaCoupler = "0.1"
+Dates = "1.9"
```
I think we can just have "1" compats for Dates and Statistics since they're in the standard library

juliasloan25 commented on test/regridder_tests.jl:
```diff
@@ -236,5 +236,79 @@ for FT in (Float32, Float64)
+            # Test setup
+            R = FT(6371e3)
+            space = TestHelper.create_space(FT, nz = 2, ne = 16, R = R)
+            # space2d = TestHelper.create_space(FT, nz = 1) # TODO
```
can we get rid of this?

juliasloan25 commented on test/regridder_tests.jl:
```diff
@@ -236,5 +236,79 @@ for FT in (Float32, Float64)
+            mkpath(REGRID_DIR)
+
+            # lat-lon dataset
+            data = ones(720, 360, 2, 3)
```
could you add a comment explaining each of these numbers please?

juliasloan25 commented on test/regridder_tests.jl:
```diff
@@ -236,5 +236,79 @@ for FT in (Float32, Float64)
+                defVar(ds, "z", z, ("z",))
+                defVar(ds, "date", time, ("date",))
+
+                defVar(ds, varname, data, ("lon", "lat", "z", "date"))
```
could you add a comment explaining why this specific order is necessary?

juliasloan25 commented on src/Regridder.jl:
```diff
@@ -179,7 +227,15 @@ function hdwrite_regridfile_rll_to_cgll(
+        length(times),
+    )
 
     # TODO: extend write! to handle time-dependent fields
```
we can remove this TODO (thanks Anna!)

juliasloan25 commented on src/Regridder.jl:
```diff
@@ -140,34 +180,42 @@ function hdwrite_regridfile_rll_to_cgll(
             @warn "No dates available in file $datafile_rll"
             data_dates = [Dates.DateTime(0)]
         end
+        if space isa Spaces.ExtrudedFiniteDifferenceSpace
```
we could have two methods of `get_coords` that dispatch on the type of `space` instead of having an `if/else` here. But I think this way is fine too :)

juliasloan25 reviewed (commented):
Looks good! I mostly requested more comments :) thank you Lenka!

LenkaNovak commented:
> Looks good! I mostly requested more comments :) thank you Lenka!

Thank you for the review, @juliasloan25 ! I've addressed your comments. Please lmk if there's anything else.

juliasloan25 reviewed (approved):
Looks good, thank you for the changes!
